src/main/java/org/folio/search/model/metadata/ResourceDescription.java

```diff
@@ -57,35 +55,12 @@
    */
   private Map<String, JsonNode> indexMappings;
 
+  /**
+   * Defined reusable types that can be referenced via $type property.
+   */
+  private Map<String, FieldDescription> fieldTypes = Collections.emptyMap();
+
   @JsonIgnore
-  @Setter(value = AccessLevel.NONE)
+  @Setter(AccessLevel.PACKAGE)
```

Comment:
Why do we need package level setter here, why it cannot be public?

Reply:
It is an calculated field that depends on `fields`, I would not allow it to be modified.

Reply:
if so, flatten fields should be collected to unmodifiable map at least to prevent further modifications

Reply:
Ah, yes, totally makes sense.